### ISSUE TITLE
Regression in 0.8.5: S3 bucket notifications no longer invoke Lambda function

### ISSUE THREAD

**Issue by GeoffreyBooth** [localstack/localstack#614]
This worked in 0.8.4, but not in 0.8.5:

```coffee
s3.putBucketNotificationConfiguration
    Bucket: 'inbox'
    NotificationConfiguration:
      LambdaFunctionConfigurations: [
        Events: ['s3:ObjectCreated:*']
        LambdaFunctionArn: 'arn:aws:lambda:us-east-1:000000000000:function:inboxHandler'
        Id: 'inboxHandler'
      ]
  , (err, data) ->
```

**Comment by whummer** [comment]
Thanks for reporting @GeoffreyBooth, this should be fixed in @645 - can you please give it a try?

(If this works, can you also verify whether #605 works for you.)

**Comment by GeoffreyBooth** [comment]
Hi @whummer I tried to test this, but when I run `make docker-build` I can’t get the tests to pass (and therefore the build to complete). I get 9 errors, most with `localstack.services.generic_proxy: ERROR: Error forwarding request: HTTPConnectionPool(host='127.0.0.1', port=4561): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f694343cf50>: Failed to establish a new connection: [Errno 111] Connection refused',))`

**Comment by jonaskint** [comment]
An extra update is when i use single part upload with s3api it does work:
```
aws s3api --endpoint=http://localhost:4572  put-object --body ~/Desktop/DSC_0004.JPG --bucket=assets --key=test.jpg
```
```
test_localstack | 2018-07-23T09:30:57:DEBUG:localstack.services.awslambda.lambda_executors: Running lambda cmd: CONTAINER_ID="$(docker create  -e AWS_LAMBDA_FUNCTION_INVOKED_ARN="$AWS_LAMBDA_FUNCTION_INVOKED_ARN" -e LOG_LEVEL="$LOG_LEVEL" -e SENTRY_DNS="$SENTRY_DNS" -e AWS_LAMBDA_EVENT_BODY="$AWS_LAMBDA_EVENT_BODY" -e HOSTNAME="$HOSTNAME" -e ENVIRONMENT="$ENVIRONMENT" -e AWS_LAMBDA_FUNCTION_VERSION="$AWS_LAMBDA_FUNCTION_VERSION" -e LOCALSTACK_HOSTNAME="$LOCALSTACK_HOSTNAME" -e AWS_LAMBDA_FUNCTION_NAME="$AWS_LAMBDA_FUNCTION_NAME" "lambci/lambda:go1.x" "main")";docker cp "/tmp/localstack/zipfile.4aad65c9/." "$CONTAINER_ID:/var/task";docker start -a "$CONTAINER_ID";
test_localstack | 2018-07-23T09:30:57:DEBUG:localstack.services.awslambda.lambda_executors: Lambda result / log output:
test_localstack | {"asynchronous": "True"}
test_localstack | >Lambda executed asynchronously
```

**Comment by jonaskint** [comment]
I think it might have something to do with: https://github.com/localstack/localstack/blob/master/localstack/services/s3/s3_listener.py#L504

**Comment by dknepprath** [comment]
@jonaskint Thanks for leaving such a good breadcrumb on this. I recreated your issue and have a branch with the fix. Will add some unit tests and get a PR up this week.

**Comment by whummer** [comment]
Closing this issue, should be fixed in #1051. Please open a follow-up issue if the problem persists. Thanks
